Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp

```diff
@@ -240,13 +241,26 @@ void Wiimote::HandleRequestStatus(const OutputReportRequestStatus&)
   // Max battery level seems to be 0xc8 (decimal 200)
   constexpr u8 MAX_BATTERY_LEVEL = 0xc8;
 
-  m_status.battery = u8(std::lround(m_battery_setting.GetValue() / 100 * MAX_BATTERY_LEVEL));
-
+#ifdef SYSTEM_BATTERY_LEVEL_AVAILABLE
+  std::optional<float> battery;
+#endif
   if (Core::WantsDeterminism())
   {
     // One less thing to break determinism:
     m_status.battery = MAX_BATTERY_LEVEL;
   }
+#ifdef SYSTEM_BATTERY_LEVEL_AVAILABLE
+  else if (m_system_battery_setting.GetValue() && (battery = GetSystemBatteryLevel()))
```

Comment:
Yes, but it is most elegant if written this way. I don't like that it would otherwise fetch the system battery even if you turned that feature off. Alternatively, I would have to add another m_system_battery_setting_check.